tlair/src/main/scala/at/forsyte/apalache/tla/lir/TlaType1.scala

```diff
@@ -37,6 +39,22 @@ object TlaType1 {
       case _ => throw new TypingException("Expected Typed(_: TlaType1), found: " + typeTag, UID.nullId)
     }
   }
+
+  // returns true iff the type is a monotype, i.e. if it contains on type variables
+  def isMono(tt: TlaType1): Boolean = tt match {
+    case _: VarT1                 => false
```

Comment:
Why doesn't this just using the constructor pattern match?

```suggestion
    case VarT1(_)                 => false
```

Reply:
Because they're semantically equivalent, but the `_: X` approach generalizes better with arity.

Reply:
Doesn't seem clear to me that the generalization helps with anything here, but fine with leaving it if you prefer.